lib/dfe/analytics/entity_table_check_job.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module DfE
+  module Analytics
+    # Reschedules to run every 24hours
+    class EntityTableCheckJob < AnalyticsJob
+      WAIT_TIME = Date.tomorrow.midnight
+
+      def perform
+        DfE::Analytics.entities_for_analytics.each do |entity_name|
+          DfE::Analytics.models_for_entity(entity_name).each do |model|
+            entity_table_check_event = DfE::Analytics::Event.new
+                                                            .with_type('entity_table_check')
+                                                            .with_entity_table_name(model.table_name)
+                                                            .with_data(entity_table_check_data(model))
+                                                            .as_json
+            DfE::Analytics::SendEvents.perform_later([entity_table_check_event])
+            Rails.logger.info("Processing data for #{model.table_name} with row count #{model.count}")
+          end
+        end
+      ensure
+        reschedule_job
+      end
+
+      def entity_table_check_data(model)
+        {
+          number_of_rows: model.count,
+          checksum: checksum(model)
+        }
+      end
+
+      def reschedule_job
+        self.class.set(wait_until: WAIT_TIME).perform_later
+      end
+
+      def checksum(model)
+        table_data = model.order(id: :asc)
```

Comment:
@ericaporter is there any risk that between the time this logic is triggered and the time the checksum is calculated, changes could be made to the underlying table? If so to reduce this risk would it be worth filtering the ids by ```created_at <= trigger time``` ...? That wouldn't handle deletions though but they're much rarer.

Reply:
Really good point. I've updated the method to chain the order, the pluck and the join which will reduce the time between building and executing this query. It might be possible to go further than this and to lock it in a transaction say, will think about whether this would be helpful here

Reply:
Now updated to use timestamp_calculated_at to reduce risk of this